cloud/services/compute/subnets/reconcile.go

```diff
@@ -43,9 +45,24 @@ func (s *Service) Reconcile(ctx context.Context) error {
 func (s *Service) Delete(ctx context.Context) error {
 	logger := log.FromContext(ctx)
 	for _, subnetSpec := range s.scope.SubnetSpecs() {
-		logger.V(2).Info("Deleting a subnet", "name", subnetSpec.Name)
 		subnetKey := meta.RegionalKey(subnetSpec.Name, s.getSubnetRegion(subnetSpec))
-		err := s.subnets.Delete(ctx, subnetKey)
+		logger.V(2).Info("Looking for subnet before deleting", "name", subnetSpec.Name)
+		subnet, err := s.subnets.Get(ctx, subnetKey)
+		if err != nil && !gcperrors.IsNotFound(err) {
+			logger.Error(err, "Error getting subnet", "name", subnetSpec.Name)
+			return err
+		} else if gcperrors.IsNotFound(err) {
+			continue
+		}
+
+		// Skip delete if subnet was not created by CAPG.
+		// If subnet description is not set by the Spec, or by our default value, then assume it was created externally.
+		if subnet.Description != infrav1.ClusterTagKey(s.scope.Name()) && (subnetSpec.Description == "" || subnet.Description != subnetSpec.Description) {
+			return nil
```

Comment:
I'd add some logging also here
```suggestion
			logger.V(2).Info("Skipping subnet deletion as it was created outside of Cluster API", "name", subnetSpec.Name)
			return nil
```